internal/cli/status.go

```diff
@@ -0,0 +1,845 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/dustin/go-humanize"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/posener/complete"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clicontext"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type StatusCommand struct {
+	*baseCommand
+
+	flagContextName string
+	flagVerbose     bool
+	flagJson        bool
+	flagAllProjects bool
+	filterFlags     filterFlags
+
+	serverCtx *clicontext.Config
+}
+
+func (c *StatusCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithConfig(true), // optional config loading
+	); err != nil {
+		return 1
+	}
+
+	var ctxName string
+	defaultName, err := c.contextStorage.Default()
+	if err != nil {
+		c.ui.Output(
+			"Error getting default context: %s",
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+	ctxName = defaultName
+
+	ctxConfig, err := c.contextStorage.Load(ctxName)
+	if err != nil {
+		c.ui.Output("Error loading context %q: %s", ctxName, err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	c.serverCtx = ctxConfig
+
+	cmdArgs := flagSet.Args()
+
+	if len(cmdArgs) > 1 {
+		c.ui.Output("No more than 1 argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Determine which view to show based on user input
+	var projectTarget, appTarget string
+	if len(cmdArgs) >= 1 {
+		s := cmdArgs[0]
+		target := strings.Split(s, "/") // This breaks if we allow projects with "/" in the name
```

Comment:
We do allow projects with `/` in the name. I would support removing this ability and requiring the user to specify the app via the `-app` flag, rather than having two ways of doing that.